### PR TITLE
feat(gatsby-transformer-yaml): support typeName

### PR DESCRIPTION
**Pull request by MeLlamoPablo**
## Description

This PR adds support for the `typeName` option to `gatsby-transformer-yaml`, as is implemented in `gatsby-transformer-json`

## Related Issues

None

### CODE REVIEW

File: packages/gatsby-transformer-yaml/README.md [hunk 1]
```diff
@@ -164,3 +164,85 @@ Which would return:
...
+      }
+    }
+  }
+}
+```
+
+or a function that receives the following arguments:
+
+- `node`: the graphql node that is being processed, e.g. a File node with
+  json content
```
Comment (wardpeet) [inline packages/gatsby-transformer-yaml/README.md]:
```suggestion
  yaml content
```

File: packages/gatsby-transformer-yaml/README.md [hunk 2]
```diff
@@ -164,3 +164,85 @@ Which would return:
...
+- `object`: a single object (either an item from an array or the whole json content)
```
Comment (wardpeet) [inline packages/gatsby-transformer-yaml/README.md]:
```suggestion
- `object`: a single object (either an item from an array or the whole yaml content)
```

File: packages/gatsby-transformer-yaml/README.md [hunk 3]
```diff
@@ -164,3 +164,85 @@ Which would return:
...
+- `isArray`: boolean, true if `object` is part of an array
+
+```json
```
Comment (wardpeet) [inline packages/gatsby-transformer-yaml/README.md]: You'll need to move to yaml
```yaml
---
- level: info
  message: hurray
- level: info
  message: it works
- level: warning
  message: look out
```

**Comment by MeLlamoPablo** [comment]
@wardpeet thanks for the quick response! I pretty much copy-pasted the `README` and forgot those, sorry for that :sweat_smile:

**Pull request merged**